src/languageSupport/languages/flux/lsp/worker/flux.worker.ts

```diff
@@ -1,4 +1,89 @@
-import Buffer from 'src/languageSupport/languages/flux/lsp/worker/buffer'
+import {Lsp} from '@influxdata/flux-lsp-browser'
+import {Methods} from 'src/languageSupport/languages/flux/lsp/utils'
+
+const respond = (msg, cb) => {
+  try {
+    // LSP spec permits a blank response
+    if (!msg) {
+      return
+    }
+    const d = JSON.parse(msg)
+    cb(d)
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+class Buffer {
+  private _i = 0
+  private _o = 0
+  private _initBuffer = new Array(100)
+  private _serverStartSignaled = false
+  private _initialized = false // ACK received from monaco
+  private _server: Lsp
+
+  private _insertBuffer = req => {
+    this._initBuffer[this._i % 100] = req
+    this._i++
+  }
+
+  private _consumeInitBuffer = async cb => {
+    let msg = this._initBuffer[this._o % 100]
+    do {
+      const res = await this._server.send(JSON.stringify(msg))
+      respond(res, cb)
+      this._o++
+      msg = this._initBuffer[this._o % 100]
+    } while (!!msg)
+    return
+  }
+
+  send(data, cb) {
+    // Handle race condition, because LSP is lazy loaded while init messages are received.
+    switch (
+      `${data.method}|${this._serverStartSignaled}|${this._initialized}`
+    ) {
+      case `${Methods.Initialize}|false|false`:
+        this._serverStartSignaled = true
+        const initMsg = data
+        start(cb).then(s => {
+          this._server = s
+          this._server
+            .send(JSON.stringify(initMsg))
+            .then(res => respond(res, cb))
+        })
+        break
+      case `${Methods.Initialize}|true|false`: // initialize already received
+        break
+      case `${Methods.Initialized}|true|false`:
+      case `${Methods.Initialized}|true|true`:
+        this._server.send(JSON.stringify(data)).then(async _ => {
+          await this._consumeInitBuffer(cb)
+          this._initialized = true
+        })
+        break
+      default:
+        if (!this._initialized) {
```

Comment:
This is only moved code. I'd rather not invest much into it, [because we do need to remove it](https://github.com/influxdata/ui/issues/5859).

Reply:
Yes, more comments would have been useful. There was a long PR convo at the time, and I was asked to keep the explanation convo in the PR (not code comments).

Boot series:
1. monaco-editor sends `initialize`
2. monaco-editor sends `didOpen` message
3. LSP comes online
4. LSP returns response to `initialize`
5. monaco-editor then sends `initialized`
6. LSP responds with `initialized`
7. then, and only then, can the LSP respond to the `didOpen` message (from earlier) without error.


The LSP server worker and monaco-editor worker are booted in parallel. Since the monaco-editor boots up much faster, it starts spamming the LSP with repeated `initialize` messages. I could have just enqueued all the duplicate messages in the ring buffer...but decided instead to ignore repeated messages.

Additionally, the ring buffer enqueues all the monaco-editor request messages in the proper order. This prevents the LSP server from errorring when it finally does come online.

Reply:
We can add that^^ as comments in the code. Up to Paul.